static/src/js/components/AuthRequiredLayout/AuthRequiredLayout.jsx

```diff
@@ -16,20 +16,25 @@ const AuthRequiredLayout = () => {
 
   const location = useLocation()
 
-  // If the app is still loading the auth context, return null to 'wait'
-  if (authLoading) {
-    return null
-  }
-
-  const isExpired = isTokenExpired(tokenExpires)
-
-  // If we have a token value that has expired, redirect to login the user again
-  if (isExpired) {
-    const nextPath = location.pathname + location.search
-
-    window.location.href = `${apiHost}/saml-login?target=${encodeURIComponent(nextPath)}`
-
-    return null
+  useEffect(() => {
+    // If we have a token value that has expired, redirect to login the user again
+    if (isTokenExpired(tokenExpires)) {
```

Comment:
I think maybe I should say `if (!authLoading && isTokenExpired(tokenExpires))`  just to be safe.